nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -1862,63 +2073,63 @@ impl<'a> Planner<'a> {
         zone_kind: ZoneKind,
         mgs_updates: &PlanningMgsUpdatesStepReport,
     ) -> Result<BlueprintZoneImageSource, TufRepoContentsError> {
-        let source_repo =
-            if self.is_zone_ready_for_update(zone_kind, mgs_updates)? {
-                self.input.tuf_repo().description()
-            } else {
-                self.input.old_repo().description()
-            };
+        let source_repo = if self.is_zone_ready_for_update(mgs_updates)? {
+            self.input.tuf_repo().description()
+        } else {
+            self.input.old_repo().description()
+        };
         source_repo.zone_image_source(zone_kind)
     }
 
-    /// Return `true` iff a zone of the given kind is ready to be updated;
-    /// i.e., its dependencies have been updated.
+    /// Return `true` iff a zone is ready to be updated; i.e., its dependencies
+    /// have been updated.
     fn is_zone_ready_for_update(
         &self,
-        zone_kind: ZoneKind,
         mgs_updates: &PlanningMgsUpdatesStepReport,
     ) -> Result<bool, TufRepoContentsError> {
```

Comment:
I think this could just return `bool` now, except I'm worried you do still need some logic here to avoid deploying new Nexus zones before the rest of the system has been updated.  I'm not _sure_ that's necessary?  But it seems safer and I assumed it was what we'd keep doing.